### PR TITLE
Unhardcode bionics that can't be removed.

### PR DESCRIPTION
**Pull request by Ramza13**
#### Summary
None

#### Purpose of change
Unhardcode bionics that can't be removed. 

#### Describe the solution
Remove the hardcoded exceptions in can_uninstall_bionic and add new json member to achieve same effect.

### CODE REVIEW

**Comment by actual-nh** [comment]
Can these be done with positional parameters using JSON (like `%1$s` and `%2$s`)? That way, translators can rearrange them if need be (admittedly, the original hardcoded versions didn't have them, but that's actually a bug in them).

**Comment by Ramza13** [comment]
> 
> 
> Can these be done with positional parameters using JSON (like `%1$s` and `%2$s`)? That way, translators can rearrange them if need be (admittedly, the original hardcoded versions didn't have them, but that's actually a bug in them). Since it's a new JSON field, it may need to be read in by a different mechanism; see doc/TRANSLATING.md.

Just did this assuming that I'm understanding you correctly.  Didn't seem like it needed a different mechanism.

**Comment by actual-nh** [comment]
> Just did this assuming that I'm understanding you correctly. Didn't seem like it needed a different mechanism.

Thanks! Re any different mechanism needed b/c a new to-be-translated JSON field - @BrettDong? Any modifications to, say, extract_json_strings.py needed?

File: src/bionics.cpp [hunk 1]
```diff
@@ -2131,9 +2126,8 @@ bool Character::can_uninstall_bionic( const bionic_id &b_id, player &installer,
         }
     }
 
-    if( b_id == bio_eye_optic ) {
-        popup( _( "The Telescopic Lenses are part of %s eyes now.  Removing them would leave %s blind." ),
-               disp_name( true ), disp_name() );
+    if( !b_id->cant_remove_reason.empty() ) {
```
Comment (anothersimulacrum) [inline src/bionics.cpp:2129]: Maybe it would be better to use a cata::optional here?

File: src/bionics.h [hunk 2]
```diff
@@ -33,6 +33,8 @@ struct bionic_data {
 
     translation name;
     translation description;
+
+    cata::optional<std::string> cant_remove_reason;
```
Comment (Qrox) [inline src/bionics.h:37]:
```suggestion
    cata::optional<translation> cant_remove_reason;
```

File: src/bionics.cpp [hunk 3]
```diff
@@ -2131,9 +2126,8 @@ bool Character::can_uninstall_bionic( const bionic_id &b_id, player &installer,
         }
     }
 
-    if( b_id == bio_eye_optic ) {
-        popup( _( "The Telescopic Lenses are part of %s eyes now.  Removing them would leave %s blind." ),
-               disp_name( true ), disp_name() );
+    if( b_id->cant_remove_reason.has_value() ) {
+        popup( string_format( _( b_id->cant_remove_reason.value() ), disp_name( true ), disp_name() ) );
```
Comment (Qrox) [inline src/bionics.cpp:2130]:
```suggestion
        popup( string_format( b_id->cant_remove_reason.value(), disp_name( true ), disp_name() ) );
```

File: data/json/bionics.json [hunk 4]
```diff
@@ -420,6 +420,7 @@
     "description": "Much of the material in your inner eye has been surgically removed and replaced with an array of high-powered, auto-focusing lenses.  They function as integrated binoculars, letting you see much farther than before, and any vision problems you might have had are now gone.",
     "occupied_bodyparts": [ [ "eyes", 2 ] ],
     "flags": [ "ENHANCED_VISION" ],
+    "cant_remove_reason": "The Telescopic Lenses are part of %1$s eyes now. Removing them would leave %2$s blind.",
```
Comment (ZhilkinSerg) [inline data/json/bionics.json:423]:
```suggestion
    "cant_remove_reason": "The Telescopic Lenses are part of %1$s eyes now.  Removing them would leave %2$s blind.",
```

**Pull request merged**